pkg/noderesourcetopology/cache/foreign_pods.go

```diff
@@ -72,7 +74,7 @@ func IsForeignPod(pod *corev1.Pod) bool {
 		// nothing to do here - we know already about this pod
 		return false
 	}
-	return true
+	return resourcerequests.AreExclusiveForPod(pod)
```

Comment:
I need to doublecheck how this plays out with PFP compute method "all"

Reply:
it will addressed in the next upload. Spoiler: adding a config option to let the user control the behavior. This seems to be the only meaningful way to interact with the other changes in this PR.